src/Illuminate/Foundation/Testing/Constraints/CountInDatabase.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Illuminate\Foundation\Testing\Constraints;
+
+use Illuminate\Database\Connection;
+use PHPUnit\Framework\Constraint\Constraint;
+
+class CountInDatabase extends Constraint
+{
+    /**
+     * Number of records that will be shown in the console in case of failure.
+     *
+     * @var int
+     */
+    protected $show = 3;
+
+    /**
+     * The database connection.
+     *
+     * @var \Illuminate\Database\Connection
+     */
+    protected $database;
+
+    /**
+     * The data that will be used to narrow the search in the database table.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * The expected count the search will produce in database.
+     *
+     * @var int
+     */
+    protected $count;
+
+    /**
+     * Create a new constraint instance.
+     *
+     * @param  \Illuminate\Database\Connection $database
+     * @param  array $data
+     * @param  int $count
+     * @return void
+     */
+    public function __construct(Connection $database, array $data, int $count)
```

Comment:
Please don't use scalar type annotations in the core. Only in the phpdoc please.